### PR TITLE
Reordering slides

### PR DESCRIPTION
**Pull request by gregdolley**

### CODE REVIEW

File: __tests__/reorder-slide.test.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            let pptx = new PPTX.Composer();
+
+            await pptx.load(`${__dirname}/fixtures/three-slides.pptx`);
+            await pptx.compose(async pres => {
+                let slide = pres.getSlide(3);
+                slide.moveTo(2);
+
+                slide.addText(text => {
+                    text
+                        .value('If this slide says "original slide #3" but is in slide position 2, then it worked.')
```
Comment (heavysixer) [inline __tests__/reorder-slide.test.js]: Are you manually validating that the slide text is correct? Is there no way to read the values of the slide?

File: __tests__/reorder-slide.test.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+            await pptx.save(`${tmpDir}/${testFileName}`);
+            expect(fs.existsSync(`${tmpDir}/${testFileName}`)).toBe(true);
+        } catch (err) {
+            console.warn(err);
+            throw err;
+        }
+    });
+
+    test('should be able to load an existing PPTX (containing six slides) and move slide #5 to slide #2', async () => {
```
Comment (heavysixer) [inline __tests__/reorder-slide.test.js]: What is the difference between this test and the previous one?

File: __tests__/reorder-slide.test.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+            await pptx.save(`${tmpDir}/${testFileName}`);
+            expect(fs.existsSync(`${tmpDir}/${testFileName}`)).toBe(true);
+        } catch (err) {
+            console.warn(err);
+            throw err;
+        }
+    });
+
+    test('should be able to load an existing PPTX (containing six slides) and move slide #2 to slide #6', async () => {
```
Comment (heavysixer) [inline __tests__/reorder-slide.test.js]: what is the difference between this test and the previous two?

File: __tests__/reorder-slide.test.js (hunk 1 above)
Comment (gregdolley) [reply __tests__/reorder-slide.test.js]: Technically, yes, we could load up an already correct version of the file into an XML string, and just compare the XML contents, here's the problem: as we add additional features to node-pptx, the XML changes, and then that invalidates the test. I've been holding off on doing direct comparisons like this because I don't want to end up changing a bunch of tests after every new feature.

File: __tests__/reorder-slide.test.js (hunk 2 above)
Comment (gregdolley) [reply __tests__/reorder-slide.test.js]: The first one tests an edge-case, where the slides are right next to each other, and one of the slides being swapped is the last one (also, that test moves a slide downward). The second test moves slide #5 upward in the stack. The direction of the move executes different parts of code.

File: __tests__/reorder-slide.test.js (hunk 3 above)
Comment (gregdolley) [reply __tests__/reorder-slide.test.js]: Moves a slide downward, and also hits the last slide in the stack.

File: __tests__/reorder-slide.test.js (hunk 1 above)
Comment (heavysixer) [reply __tests__/reorder-slide.test.js]: Ok understood, but if you can "select" a slide to modify can't you check it's content?

File: __tests__/reorder-slide.test.js (hunk 3 above)
Comment (heavysixer) [reply __tests__/reorder-slide.test.js]: 👍

File: __tests__/reorder-slide.test.js (hunk 2 above)
Comment (heavysixer) [reply __tests__/reorder-slide.test.js]: 👍

File: __tests__/reorder-slide.test.js (hunk 1 above)
Comment (gregdolley) [reply __tests__/reorder-slide.test.js]: You _could_ load a correct file, pick out that slide, and do a text comparison, yeah. I could add that if you want...

Comment (heavysixer) [reply __tests__/reorder-slide.test.js]: I think that is the correct way to test this functionality thanks!

Comment (gregdolley) [reply __tests__/reorder-slide.test.js]: Done. See new stuff in reorder-slide.test.js.
